opennms-webapp-rest/src/main/java/org/opennms/web/rest/v1/TopologyRestService.java

```diff
@@ -0,0 +1,142 @@
+
+package org.opennms.web.rest.v1;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.core.Response;
+
+
+
+import org.graphdrawing.graphml.GraphmlType;
+import org.opennms.features.graphml.model.GraphML;
+import org.opennms.features.graphml.model.GraphMLEdge;
+import org.opennms.features.graphml.model.GraphMLGraph;
+import org.opennms.features.graphml.model.GraphMLNode;
+import org.opennms.features.graphml.model.GraphMLWriter;
+import org.opennms.features.graphml.model.InvalidGraphException;
+import org.opennms.netmgt.dao.api.TopologyDao;
+import org.opennms.netmgt.model.OnmsTopology;
+import org.opennms.netmgt.model.topology.Topology;
+import org.opennms.netmgt.model.topology.Topology.ProtocolSupported;
+import org.springframework.stereotype.Component;
+
+@Component
+@Path("topology")
+public class TopologyRestService {
+/*
+    private final static String ID = "id";
+    private final static String DESCRIPTION = "description";
+    private final static String IP_ADDRESS = "ipAddr";
+    private final static String LOCKED = "locked";
+    private final static String FOREIGN_SOURCE = "foreignSource";
+    private final static String FOREIGN_ID = "foreignID";
+    private final static String SELECTED = "selected";
+    private final static String STYLE_NAME = "styleName";
+    private final static String X = "x";
+    private final static String Y = "y";
+    private final static String PREFERRED_LAYOUT = "preferred-layout";
+    private final static String FOCUS_STRATEGY = "focus-strategy";
+    private final static String FOCUS_IDS = "focus-ids";
+    private final static String SEMANTIC_ZOOM_LEVEL = "semantic-zoom-level";
+    private final static String VERTEX_STATUS_PROVIDER = "vertex-status-provider";
+    private final static String LEVEL = "level";
+    private final static String EDGE_PATH_OFFSET = "edge-path-offset";
+    private final static String BREADCRUMB_STRATEGY = "breadcrumb-strategy";
+  */  
+    private final static String TOPOLOGY_NAMESPACE_PREFIX="Topology::";
+    private final static String NAMESPACE = "namespace";
+    private final static String ICON_KEY = "iconKey";
+    private final static String LABEL = "label";
+    private final static String NODE_ID = "nodeID";
+    private final static String TOOLTIP_TEXT = "tooltipText";
+//FIXME support port
+//    private final static String SOURCE_PORT= "sourceport";
+//    private final static String TARGET_PORT= "targetport";
+    private final static String SOURCE_IFINDEX= "sourceifindex";
+    private final static String TARGET_IFINDEX= "targetifindex";
+    
+    private TopologyDao m_topologyDao;
+
+
+    @GET
+    @Path("{supported-protocol}")
+    public Response getGraph(@PathParam("supported-protocol") String supportedprotocol) throws InvalidGraphException {
+        if (supportedprotocol.equals(Topology.ProtocolSupported.CDP.name())) {
```

Comment:
This is not very generic. 
Depending on how generic the solution is, we either need to iterate over the above mentioned `TopologyProvider` and determine if the provider can provide the protocol. 
Otherwise I'd prefer something like

```java
ProtocolSupported protocol = ProtocolSupported.valueOf(supportedProtocol);
GraphMLType graph = getGraph(protocol);
return Response.ok(graph).build();
```

Reply:
Yup